src/librustc_codegen_llvm/back/link.rs

```diff
@@ -462,6 +462,26 @@ fn print_native_static_libs(sess: &Session, all_native_libs: &[NativeLibrary]) {
     }
 }
 
+fn get_file_path(sess: &Session, name: &str) -> PathBuf {
+    let fs = sess.target_filesearch(PathKind::Native);
+    let mut paths = Vec::new();
```

Comment:
Only the last element of `paths` is ever used, correct? In that case, I'd keep this as an `Option` of the last path found.

Relatedly, is there a reason for preferring the last item in `for_each_lib_search_path` rather than preferring, say, the first item found, or even preferring the result inside `fs.get_lib_path()`, which is the current default? If users happened to have files in their lib search path that matched the name of the searched-for file but had previously been picking up the item in the sysroot, this change will move them to depending on the value from `for_each_lib_search_path`, yes?